app/add_some_abilities/add_protection.py

```diff
@@ -0,0 +1,7 @@
+def protection(knight: dict) -> None:
+    knight["add_some_abilities"] = armour_sum(knight["armour"])
+
+
+def armour_sum(armour: dict) -> int:
+    return sum([part_of_armour["add_some_abilities"]
+                for part_of_armour in armour])

```

Comment:
The list comprehension is correctly used here, but the variable name `part_of_armour["add_some_abilities"]` might be misleading. Ensure that `"add_some_abilities"` is the correct key for accessing the armour value. If not, update it to the correct key.